packages/flutter/lib/src/painting/images.dart

```diff
@@ -117,6 +117,15 @@ class DecorationImage {
   /// in the top right.
   final bool matchTextDirection;
 
+  /// Creates a [DecorationImagePainter] for this [DecorationImage].
+  ///
+  /// The `onChanged` argument must not be null. It will be called whenever the
```

Comment:
Maybe then `assert` that `onChanged` is not null?

Reply:
done